app/jobs/scrape_job.rb

```diff
@@ -0,0 +1,66 @@
+class PostScrapeWorker < ActiveJob::Base
+  queue_as :default
+
+  require 'capybara'
+  require 'capybara/poltergeist'
+
+  include Sidekiq::Worker
+  include Capybara::DSL
+  Capybara.register_driver :poltergeist do |app|
+    Capybara::Poltergeist::Driver.new(app, timeout: 60, js_errors: false)
+  end
+  Capybara.default_driver = :poltergeist
+
+  def perform(new_catalogue)
+    new_catalogue.each do |catalogue_num|
+      begin
+
+        #visit page
+        visit "http://salefinder.com.au/woolworths-catalogue/weekly-specials-catalogue/#{catalogue_num}/catalogue"
+        
+        #creating new catalogue
+        @catalogue = Catalogue.new
+        @catalogue.title = page.all('div#breadcrumb li a')[2].text
+        @catalogue.date = page.find('span.sale-dates').text
+        @catalogue.catalogue_num = catalogue_num
+        @catalogue.shop = page.all('div#breadcrumb li a')[1].text
+        @catalogue.shop.slice!(" catalogues")
+        if @catalogue.shop == "Woolworths"

```

Comment:
I agree that having a ```shop``` model would model the ideal relationship in this data. When I built this, I'm looking for the minimum model to get it running and also in this program, we only have 2 shops. so at this point of time, I decided to stick without the ```shop``` model, and implement it later in the future.